src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,48 +1,45 @@
 // (C) 2020 GoodData Corporation
 import * as React from "react";
 import without = require("lodash/without");
-import { Execution } from "@gooddata/typings";
-import { stringToFloat } from "../../../helpers/utils";
+import cx from "classnames";
 import { IntlWrapper } from "../../core/base/IntlWrapper";
 import { IntlTranslationsProvider, ITranslationsComponentProps } from "../../core/base/TranslationsProvider";
 import PushpinSizeLegend from "./legends/PushpinSizeLegend";
-import { getGeoData, getFormatFromExecutionResponse } from "../../../helpers/geoChart";
-import { IGeoConfig, IGeoData } from "../../../interfaces/GeoChart";
+import { IGeoConfig } from "../../../interfaces/GeoChart";
 import { TOP } from "../../visualizations/chart/legend/PositionTypes";
-import { isTwoDimensionsData } from "../../../helpers/executionResultHelper";
+import { IColorLegendItem } from "../../visualizations/typings/legend";
+import ColorLegend from "../../visualizations/chart/legend/ColorLegend";
 
 export interface IGeoChartLegendRendererProps {
     config: IGeoConfig;
-    execution: Execution.IExecutionResponses;
+    sizeData?: number[];
+    sizeFormat?: string;
+    colorData?: IColorLegendItem[];
+    colorFormat?: string;
+    isColorLegendVisible?: boolean;
     locale: string;
     position?: string;
 }
 export default function GeoChartLegendRenderer(props: IGeoChartLegendRendererProps): JSX.Element {
-    if (!props.execution) {
-        return null;
-    }
     const {
-        execution: { executionResult, executionResponse },
-        config: { mdObject: { buckets = [] } = {} },
         locale,
         position = TOP,
+        sizeData,
+        sizeFormat,
+        colorData,
+        colorFormat,
+        isColorLegendVisible = false,
     } = props;
-    const geoData: IGeoData = getGeoData(buckets, executionResponse.dimensions);
-    const { data } = executionResult;
-    const { size } = geoData;
-    const classes = `geo-legend s-geo-legend position-${position}`;
-
-    if (!size || !isTwoDimensionsData(data)) {
+    const isLegendHidden = !isColorLegendVisible && !sizeData;
+    if (isLegendHidden) {
         return null;
     }
+
+    const classes = cx("geo-legend s-geo-legend", `position-${position}`, !!sizeData && "has-size-legend");
```

Comment:
it could be
```
const hasSizeData = Boolean(sizeData);
const isLegendHidden = isColorLegendVisible || hasSizeData;

if (isLegendHidden) {
        return null;
    }

const classes = cx(
         "geo-legend s-geo-legend",
         `position-${position}`, 
         { "has-size-legend": hasSizeData }
);
```